server/src/test/java/org/elasticsearch/snapshots/SnapshotsServiceTests.java

```diff
@@ -199,6 +214,76 @@ public void testSuccessfulSnapshot() {
+        final AtomicBoolean createdSnapshot = new AtomicBoolean();
+
+        final AdminClient masterAdminClient = masterNode.client.admin();
+        masterNode.client.admin().cluster().preparePutRepository(repoName)
+            .setType(FsRepository.TYPE).setSettings(Settings.builder().put("location", randomAlphaOfLength(10)))
+            .execute(
+                assertNoFailureListener(
+                    () -> masterNode.client.admin().indices().create(
+
+                        new CreateIndexRequest(index).waitForActiveShards(ActiveShardCount.ALL)
+                            .settings(defaultIndexSettings(shards)),
+                        assertNoFailureListener(
+                            () -> {
+                                for (int i = 0; i < randomIntBetween(0, dataNodes); ++i) {
+                                    scheduleNow(this::disconnectRandomDataNode);
+                                }
+                                if (randomBoolean()) {
+                                    scheduleNow(() -> testClusterNodes.clearNetworkDisruptions());
+                                }
+                                masterAdminClient.cluster().prepareCreateSnapshot(repoName, snapshotName)
+                                    .execute(assertNoFailureListener(() -> {
+                                        for (int i = 0; i < randomIntBetween(0, dataNodes); ++i) {
+                                            scheduleNow(this::disconnectOrRestartDataNode);
+                                        }
+                                        final boolean disconnectedMaster = randomBoolean();
+                                        if (disconnectedMaster) {
+                                            scheduleNow(this::disconnectOrRestartMasterNode);
+                                        }
+                                        if (disconnectedMaster || randomBoolean()) {
+                                            scheduleSoon(() -> testClusterNodes.clearNetworkDisruptions());
+                                        } else if (randomBoolean()) {
+                                            scheduleNow(() -> testClusterNodes.clearNetworkDisruptions());
+                                        }
+                                        createdSnapshot.set(true);
+                                    }));
+                            }))));
+
+        runUntil(() -> {
+            final Optional<TestClusterNode> randomMaster = testClusterNodes.randomMasterNode();
+            if (randomMaster.isPresent()) {
+                final SnapshotsInProgress snapshotsInProgress = randomMaster.get().clusterService.state().custom(SnapshotsInProgress.TYPE);
+                return snapshotsInProgress != null && snapshotsInProgress.entries().isEmpty();
+            }
+            return false;
+        }, TimeUnit.MINUTES.toMillis(20L));
```

Comment:
that's quite a long time. How long does this test typically take?

Reply:
Ran a few k iterations and couldn't get a run to exceed `5s` => dropped it to 1 minute (just to be safe) in https://github.com/elastic/elasticsearch/pull/37686/commits/326cd5648d7b47d09abc2a360740d4d48c52d3d4